src/prestashop-pull-request-parser.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+module.exports = class PrestashopPullRequestParser {
+
+  /**
+   * @param {Logger} logger
+   */
+  constructor (logger) {
+    if (null !== logger) {
```

Comment:
Warning: in Javascript, the null object can sometime be tricky.
For example:
```
> null == undefined
true
> null === undefined
false
```
If you want to compare if logger has any value, it will be better to check it like that:
```
if (logger) { ... }
```
Because, if logger is 'undefined', or empty, in this case `if (null !== logger) {`, it will enter the if conditions.

Reply:
@CaptainYouz but should'nt I control the states of my variables so I should control whether my variable is `null`, `undefined` or empty ?

I would say if I get `undefined` instead of `null` it's an error from me, I should not write code that behaves "in case of" ?